Plugins/Json/include/Acts/Plugins/Json/JsonHierarchicalObjectConverter.hpp

```diff
@@ -0,0 +1,183 @@
+    std::string senkey = "sensitive";
+    /// The representing idntification string
+    std::string repkey = "representing";
+    /// The name identification
+    std::string namekey = "Name";
+    /// The object key
+    std::string datakey = "Object";
+
+    /// The default logger
+    std::shared_ptr<const Logger> logger;
+    /// The name of the writer
+    std::string name = "";
+
+    /// Steering to handle sensitive data
+    bool processSensitives = true;
+    /// Steering to handle approach data
+    bool processApproaches = true;
+    /// Steering to handle representing data
+    bool processRepresenting = true;
+    /// Steering to handle boundary data
+    bool processBoundaries = true;
+    /// Steering to handle volume data
+    bool processVolumes = true;
+
+    /// Constructor
+    ///
+    /// @param lname Name of the writer tool
+    /// @param lvl The output logging level
+    Config(const std::string& lname = "JsonHierarchicalObjectConverter",
+           Logging::Level lvl = Logging::INFO)
+        : logger(getDefaultLogger(lname, lvl)), name(lname) {}
+  };
+
+  using geo_id_value = uint64_t;
+  using Representation = std::map<geo_id_value, object_t>;
+
+  /// @brief Layer representation for Json writing
+  struct LayerRep {
+    // the layer id
+    GeometryID layerID;
+    Representation sensitives;
+    Representation approaches;
+    object_t representing;
+  };
+
+  /// @brief Volume representation for Json writing
+  struct VolumeRep {
+    // The geometry id
+    GeometryID volumeID;
+    /// The namne
+    std::string volumeName;
+
+    std::map<geo_id_value, LayerRep> layers;
+    Representation boundaries;
+    object_t volume;
+  };
+
+  /// @brief Detector representation for Json writing
+  struct DetectorRep {
+    std::map<geo_id_value, VolumeRep> volumes;
+  };
+
+  /// Constructor
+  ///
+  /// @param cfg configuration struct for the reader
+  JsonHierarchicalObjectConverter(const Config& cfg);
+
+  /// Destructor
+  ~JsonHierarchicalObjectConverter() = default;
+
+  /// Convert json map to Hierarchical Object Container
+  ///
+  /// @param map The indexed Hierarchical Object in json format
+  /// @param fromJson Function that return Hierarchical Object from json
+  HierarchicalGeometryContainer<object_t> jsonToHierarchicalContainer(
+      const nlohmann::json& map,
+      std::function<object_t(const nlohmann::json&)> fromJson) const;
+
+  /// Convert json map to Hierarchical Object Container
+  ///
+  /// @param hObject The Hierarchical Object Container
+  /// @param toJson Function that convert Hierarchical Object to json
+  nlohmann::json hierarchicalObjectToJson(
+      const HierarchicalGeometryContainer<object_t>& hObject,
+      std::function<nlohmann::json(const object_t&)> toJson) const;
+
+  /// Write json map from Tracking Geometry
+  ///
+  /// @param tGeometry is the tracking geometry which contains the material
+  nlohmann::json trackingGeometryToJson(
```

Comment:
Why should this write the tracking geometry material? This class should be responsible only for converting the `HierarchicalGeometryContainer`.

Reply:
It is an error in the comment due to a hasty copy past from the material writer.